ntpd/src/daemon/config/ntp_source.rs

```diff
@@ -6,15 +6,34 @@ use std::{
     sync::{Arc, Mutex},
 };
 
+#[cfg(feature = "unstable_ntpv5")]
+use ntp_proto::NtpVersion;
 use rustls::pki_types::CertificateDer;
 use serde::{de, Deserialize, Deserializer};
 
 use super::super::keyexchange::certificates_from_file;
 
+#[cfg(feature = "unstable_ntpv5")]
+fn deserialize_ntp_version<'de, D>(deserializer: D) -> Result<Option<NtpVersion>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let version = Option::<u8>::deserialize(deserializer)?;
+    match version {
+        None => Ok(None),
+        Some(4) => Ok(Some(NtpVersion::V4)),
+        Some(5) => Ok(Some(NtpVersion::V5)),
+        Some(v) => Err(de::Error::custom(format!("Invalid ntp version {v}"))),
+    }
+}
+
 #[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct StandardSource {
     pub address: NtpAddress,
+    #[cfg(feature = "unstable_ntpv5")]
+    #[serde(default, deserialize_with = "deserialize_ntp_version")]
+    pub ntp_version: Option<NtpVersion>,
```

Comment:
I think we should probably add a test here that (a) tests the config parsing and (b) checks that the source is actually spawned with that `ProtocolVersion` specified